posthog/models/cohort/util.py

```diff
@@ -324,7 +324,10 @@ def recalculate_cohortpeople(
             "team_id": cohort.team_id,
             "new_version": pending_version,
         },
-        settings={"optimize_on_insert": 0},
+        settings={
+            "max_execution_time": 240,
```

Comment:
The task definition here doesn't have an explicit timeout:

https://github.com/PostHog/posthog/blob/9576fab1e4720a643ba5e4feec9ba83975c9c6d3/posthog/tasks/calculate_cohort.py#L49-L50

From what I can tell we don't provide any default timeout either, so this shouldn't need to be accommodated for anywhere up the stack.